### ISSUE TITLE
Very slow with large text

### ISSUE THREAD

**Issue by ropel** [ckeditor/ckeditor5-vue#42]
I've tried embedding ClassicEditor in a clean vue project: everything is OK, but when I paste a long flat text (not formatted, just 800 lines long text) the editor became veeery slow (several seconds just to type a character and see it on screen). The same text pasted in your online (not vue) examples works without problems. After deleting all the text, editing is again responsive.
Removing "v-model" tag (thinking about a slowdown keeping the model in sync) didn't help
Tried with latest Chrome on Windows and Linux
My page is as simple as that:

```
<template>
    <ckeditor :editor="editor" v-model="editorData" :config="editorConfig"></ckeditor>
</template>
<script>
import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
export default {
  name: "editortest",
  components: {},
  data: function() {
    return {
      editor: ClassicEditor,
      editorData: "<p>Rich-text editor content.</p>",
      editorConfig: {
        // The configuration of the rich-text editor.
      }
    };
  },
  methods: {}
};
</script>

<style>
</style>
```

**Comment by Mgsy** [comment]
I can confirm it. I've just loaded 800 paragraphs as initial editor content and the editor became incredibly slow. It doesn't happen in CKEditor 5 without Vue integration. 

@oleq, could you take a look at this one?

**Comment by ropel** [comment]
What should I look ? I don't find anything about vue with CKE5

Il 10/04/19 09:35, Kacper Tomporowski ha scritto:

**Comment by oleq** [comment]
I confirm the issue. The culprit is this listener https://github.com/ckeditor/ckeditor5-vue/blob/1dc60f8a887e063ef63e45a0be8105fdfa1b0d13/src/ckeditor.js#L101-L106. When there's a lot of data, it takes lot of time to emit the event. We should probably debounce it.

**Comment by Reinmar** [comment]
Dunno how it works in Vue, so sorry for a potentially stupid question – can the content be retrieved only on demand? Or does Vue implement a sort of two-way-binding so it can't?

BTW, I'd consider something like:

```js
this.$emit( 'input', () => editor.getData(), evt, editor ); 
```

So, instead of passing `data` directly in the `input` event, we'd allow someone to retrieve it if needed. Although, if the consumer would actually read it on every `input` that would change nothing.

Sorry again if I'm missing the point :D For two-way data binding debouncing seems reasonable.

**Comment by oleq** [comment]
@Reinmar I'm afraid it is impossible. However, [in the PR](https://github.com/ckeditor/ckeditor5-vue/pull/46) I managed to reduce the number of `setData()` and `getData()` calls with some caching, which significantly improved the performance. 

In the second step, I'm gonna investigate a possibility of debouncing. Stay tuned.

**Comment by oleq** [comment]
I updated the PR with a debounced `#input` event. Any feedback appreciated.

**Comment by AndrewMast** [comment]
It would be nice to know if the editor is currently being "debounced" so a form submission can be delayed. Maybe setting a variable "typing" to true? Or at least setting a "maxWait"...

**Comment by Reinmar** [comment]
Something like – `isBusy`? :D If it solves some issues, then it sounds fine to me.

**Comment by Reinmar** [comment]
BTW, perhaps we could use the `PendingActions` plugin? It's mentioned in https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/saving-data.html#autosave-feature and could help here too.

**Comment by preeteshjain** [comment]
Was this resolved? I am still experiencing this issue in Vue.

**Comment by awmartin** [comment]
I just encountered this issue today in Vue.js. I get a half- to full-second delay while typing, making for a frustrating staccato, stop-and-go experience. Is there a mitigation for this, perhaps not using CKEditor as a Vue component and just loading it as a pure native build?

**Comment by Reinmar** [comment]
@awmartin could you open a new ticket. I suppose that there must be an option to disable the two-way data binding or a smarter debounce mechanism (e.g. similar to the one used by autosave). 

Also, a related ticket in CKEditor 5 itself: ckeditor/ckeditor5#5812.

**Comment by szohaib** [comment]
Can anyone share some update on this please?

**Comment by robertfoenix** [comment]
What about @ready="onReady" --> onReady(editor) { const data = editor.getData() } and skip the v-model?

**Comment by awmartin** [comment]
I forgot to open another ticket. :-(

For context, I use a doc with about 6000 words and a lot of paragraphs and list items to test. The delay is typically 600-700ms when it appears. I think the performance of `getData` is largely correlated to the number of top-level elements.

To bypass the `v-model` pattern and remove the hesitation, I had to remove the `change:data` event [here](https://github.com/ckeditor/ckeditor5-vue/blob/master/src/ckeditor.js#L157) and remove the `modelValue` watcher. The performance returned, and I just handled saving documents manually. You don't get any Vue.js data binding though.

The other approach I tried was just increasing the debounce interval to 3000ms. It seemed to be okay except for a couple of hiccups, which I couldn't quite track down.
